runtime/compiler/env/J9KnownObjectTable.cpp

```diff
@@ -115,13 +117,22 @@ J9::KnownObjectTable::getOrCreateIndex(uintptr_t objectPointer)
 
 
 TR::KnownObjectTable::Index
-J9::KnownObjectTable::getOrCreateIndex(uintptr_t objectPointer, bool isArrayWithConstantElements)
+J9::KnownObjectTable::getOrCreateIndex(uintptr_t objectPointer, bool isArrayWithConstantElements, int32_t stableArrayRank)
    {
    TR::KnownObjectTable::Index index = self()->getOrCreateIndex(objectPointer);
    if (isArrayWithConstantElements)
       {
       self()->addArrayWithConstantElements(index);
       }
+
+   if (stableArrayRank > 0)
+      {
+      _stableArrayRanks[index] = stableArrayRank;
+      if (!_arrayWithStableElements)
+          _arrayWithStableElements = new (self()->comp()->trHeapMemory()) TR_BitVector(self()->getEndIndex(), self()->comp()->trMemory(), heapAlloc, growable);
+      _arrayWithStableElements->set(index);
```

Comment:
Oh, good idea! I wanted to avoid the array growing when checking non-existing value. But checking `i < _stableArrayRanks.size() ` should do the work. Certainly, better to remove the vector then. Thanks!